shell/platform/android/io/flutter/embedding/engine/android/FlutterFragment.java

```diff
@@ -308,6 +330,67 @@ protected String getDartEntrypointFunctionName() {
   public void onPostResume() {
     Log.d(TAG, "onPostResume()");
     flutterEngine.getLifecycleChannel().appIsResumed();
+
+    // TODO(mattcarroll): find a better way to handle the update of UI overlays than calling through
+    //                    to platformPlugin. We're implicitly entangling the Window, Activity, Fragment,
+    //                    and engine all with this one call.
+    platformPlugin.onPostResume();
+
+    // TODO(mattcarroll): consider a more abstract way to invoke this behavior. It is very strange for
+    //                    a Fragment to have a seemingly random View responsibility, but this is what
+    //                    existed in the original embedding and I don't have a good alternative yet.
+    flutterView.updateAccessibilityFeatures();
+  }
+
+  @Override
+  public void onPause() {
+    super.onPause();
+    Log.d(TAG, "onPause()");
+    flutterEngine.getLifecycleChannel().appIsInactive();
+  }
+
+  @Override
+  public void onStop() {
+    super.onStop();
+    Log.d(TAG, "onStop()");
+    flutterEngine.getLifecycleChannel().appIsPaused();
+  }
+
+  @Override
+  public void onDestroyView() {
+    super.onDestroyView();
+    Log.d(TAG, "onDestroyView()");
+    flutterView.detachFromFlutterEngine();
+  }
+
+  @Override
+  public void onDetach() {
+    super.onDetach();
+    Log.d(TAG, "onDetach()");
+
+    // Null out the platformPlugin to avoid a possible retain cycle between the plugin, this Fragment,
+    // and this Fragment's Activity.
+    platformPlugin = null;
+
+    // Destroy our FlutterEngine if we're not set to retain it.
+    if (!retainFlutterIsolateAfterFragmentDestruction()) {
+      flutterEngine.destroy();
+      flutterEngine = null;
+    }
+  }
+
+  /**
+   * Returns true if the {@link FlutterEngine} within this {@code FlutterFragment} should outlive
+   * the {@code FlutterFragment}, itself.
+   *
+   * Defaults to false. This method can be overridden in subclasses to retain the
+   * {@link FlutterEngine}.
+   */
+  // TODO(mattcarroll): consider a dynamic determination of this preference based on whether the
+  //                    engine was created automatically, or if the engine was provided manually.
+  //                    Manually provided engines should probably not be destroyed.
```

Comment:
In the iOS embedding this is a parameter to the initializer.

Reply:
There is no initializer for a `FlutterFragment` but the answer might be to take it in as a bundle argument.  The question that remains to be seen is if subclasses should be able to control this.  I believe it is likely that they should.  I need to try the API for real to get a sense of what is appropriate.